apps/src/p5lab/spritelab/blocks.js

```diff
@@ -289,10 +289,18 @@ const customInputTypes = {
         }
       }
       block.thumbnailSize = 32;
+      // Try to get the image url for this block. If we find one,
+      // initialize the field with the image and short string.
+      // Otherwise, initialize the field with the long string and a 1 pixel
+      // wide empty image.
+      const imageUrl = Blockly.getShadowedBlockImageUrl(block);
+      const width = imageUrl.length > 0 ? block.thumbnailSize : 1;
+      const firstField =
+        imageUrl.length > 0 ? block.shortString : block.longString;
       currentInputRow
-        .appendField(block.longString)
+        .appendField(firstField)
```

Comment:
When you set a field like this (with only a string, as opposed to an instance of a custom field class), Blockly treats it as a short cut and creates a `FieldLabel` with the string set as the value. So we could more perhaps more specifically refer to this as the block's label field.